execution/order_by.go

```diff
@@ -51,24 +51,24 @@ func isSorteable(x octosql.Value) bool {
 	panic("unreachable")
 }
 
-func (ob *OrderBy) Get(ctx context.Context, variables octosql.Variables, streamID *StreamID) (RecordStream, error) {
+func (ob *OrderBy) Get(ctx context.Context, variables octosql.Variables, streamID *StreamID) (RecordStream, *ExecOutput, error) {
 	tx := storage.GetStateTransactionFromContext(ctx)
 	sourceStreamID, err := GetSourceStreamID(tx.WithPrefix(streamID.AsPrefix()), octosql.MakePhantom())
 	if err != nil {
-		return nil, errors.Wrap(err, "couldn't get source stream ID")
+		return nil, nil, errors.Wrap(err, "couldn't get source stream ID")
 	}
 
-	sourceStream, err := ob.source.Get(ctx, variables, sourceStreamID)
+	sourceStream, execOutput, err := ob.source.Get(ctx, variables, sourceStreamID)
 	if err != nil {
-		return nil, errors.Wrap(err, "couldn't get underlying stream in order by")
+		return nil, nil, errors.Wrap(err, "couldn't get underlying stream in order by")
 	}
 
 	orderedStream, err := createOrderedStream(ctx, ob.expressions, ob.directions, variables, sourceStream)
 	if err != nil {
-		return nil, errors.Wrap(err, "couldn't create ordered stream from source stream")
+		return nil, nil, errors.Wrap(err, "couldn't create ordered stream from source stream")
 	}
 
-	return orderedStream, nil
+	return orderedStream, execOutput, nil
```

Comment:
Order By waits to read the whole stream. (it doesn't make much sense in streaming situations, as it doesn't return anything until it gets the whole stream and returns it as a whole). Maybe with an exception when we're grouping by event_time + something (this would need some restructure). But for now I think this should return a zero watermark.